src/charm.py

```diff
@@ -14,76 +14,63 @@
     WaitingStatus,
 )
 from ops.framework import StoredState
+from string import ascii_letters, digits
 
 logger = logging.getLogger(__name__)
 PEER = "mysql"
 
 
 class MySQLCharm(CharmBase):
     """Charm to run MySQL on Kubernetes."""
+
     _stored = StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.image = OCIImageResource(self, 'mysql-image')
-        self.framework.observe(self.on.start, self._on_start)
+        self._stored.set_default(mysql_setup={})
+        self.image = OCIImageResource(self, "mysql-image")
         self.framework.observe(self.on.config_changed, self._on_config_changed)
 
     @property
-    def cluster_size(self) -> int:
-        """Return the size of the cluster."""
-        rel = self.model.get_relation(PEER)
-        return len(rel.units) + 1 if rel is not None else 1
+    def mysql_root_password(self) -> str:
+        if "MYSQL_ROOT_PASSWORD" not in self._stored.mysql_setup:
+            password = "".join(
+                random.choice(ascii_letters + digits) for x in range(20)
```

Comment:
Should the password generation only be done by the leader ? If so, would it be a good idea to have a guard here to check that this method is being invoked by the leader ?

Reply:
You are right, it is a possible point of failure. Fixed!